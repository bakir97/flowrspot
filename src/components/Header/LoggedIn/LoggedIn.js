import React from "react";
import profileHolder from "../../../assets/images/menu-profile-holder.png";
import profileHolder2x from "../../../assets/images/menu-profile-holder@2x.png";
import profileHolder3x from "../../../assets/images/menu-profile-holder@3x.png";
import styles from "./LoggedIn.module.scss";
const LoggedIn = ({
  user: { first_name, last_name },
  openModal,
  toggleMenu,
  showMenu
}) => {
  const handleOpenModal = () => {
    window.document.body.style.overflow = "hidden";
    showMenu && toggleMenu();

    openModal("profile");
  };
  return (
    <div onClick={handleOpenModal} className={styles.container}>
      <p className={styles.text}> {`${first_name} ${last_name}`}</p>
      <img
        src={profileHolder}
        srcSet={`${profileHolder2x} 2x,${profileHolder3x} 3x`}
        alt="profile"
      />
    </div>
  );
};

export default LoggedIn;
