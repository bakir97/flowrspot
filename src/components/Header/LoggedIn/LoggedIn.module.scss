@import "../../../assets/global/styles/colors.scss";
@import "../../../assets/global/styles/font.scss";
@import "../../../assets/global/styles/mixins.scss";
.container {
  display: flex;
  @media screen and (max-width: 768px) {
    flex-direction: column;
  }
  &:hover {
    cursor: pointer;
  }
}
.text {
  @include font-medium;
  color: $light-gray;
  margin-right: 20px;
}
