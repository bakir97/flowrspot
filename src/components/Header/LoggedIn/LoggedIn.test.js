import React from "react";
import { shallow } from "enzyme";
import LoggedIn from "./LoggedIn";
describe("LoggedIn", () => {
  it("should render correctly", () => {
    const component = shallow(<LoggedIn user={{}} />);
    expect(component).toMatchSnapshot();
  });
  it("should call trigger open modal and toggle menu ", () => {
    const mockFn = jest.fn();
    const component = shallow(
      <LoggedIn openModal={mockFn} toggleMenu={mockFn} showMenu user={{}} />
    );
    component.find("div").simulate("click");
    expect(mockFn).toHaveBeenCalledTimes(2);
  });
});
