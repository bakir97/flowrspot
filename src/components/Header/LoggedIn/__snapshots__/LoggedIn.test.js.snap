// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoggedIn should render correctly 1`] = `
<div
  className="container"
  onClick={[Function]}
>
  <p
    className="text"
  >
     
    undefined undefined
  </p>
  <img
    alt="profile"
    src="menu-profile-holder.png"
    srcSet="menu-profile-holder@2x.png 2x,menu-profile-holder@3x.png 3x"
  />
</div>
`;
